Update Kingdom-of-Miscellania plugin to calculate best rewards

RyBo opened the pull request:
Calculate most profitable rewards for given day. Reward averages gathered from the OSRS Wiki page: [Managing Miscellania](https://oldschool.runescape.wiki/w/Managing_Miscellania#Rewards). 

Assumes the Kingdom can withdraw the maximum salary of 75,000 gp, 100% favour, and that you have the total amount of workers available after completing Royal Trouble.

![komupdate](https://user-images.githubusercontent.com/8285466/51563762-85a35d80-1e4a-11e9-9bfc-52227493421a.png)

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/ResourceType.java:
```diff
@@ -0,0 +1,49 @@
+import lombok.Getter;
+
+@Getter
+public enum ResourceType
```
```suggestion
enum ResourceType
```

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/Reward.java:
```diff
@@ -0,0 +1,68 @@
+import net.runelite.api.ItemID;
+
+@Getter
+public enum Reward
```
```suggestion
enum Reward
```

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/Reward.java:
```diff
@@ -0,0 +1,68 @@
+	private final double quantity;
+	private final int rewardId;
+
+	Reward(String name, ResourceType type, double quantity, int rewardId)
```
You can just annotate the enum `@AllArgsConstructor`

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/ResourceType.java:
```diff
@@ -0,0 +1,49 @@
+
+	private String type;
+
+	ResourceType(String type)
```
`@AllArgsConstructor`

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java:
```diff
@@ -42,10 +42,10 @@
-	description = "Show amount of favor when inside Miscellania",
-	tags = {"favor", "favour", "managing", "overlay"},
-	enabledByDefault = false
+		name = "Kingdom of Miscellania",
```
Use one tab for indenting

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java:
```diff
@@ -102,6 +106,40 @@ private void processInfobox()
 
+	private void calculateRewards()
+	{
+		ResourceType type = ResourceType.FARM;
```
Can you access what the current worker distribution is? It might be in varbits. If so you could show the current profit (in addition to this?) which I think people would like too.

RyBo replied:
I wasn't able to determine the distribution through varbits before but I'm going to check again.

RyBo replied:
You absolutely can, I'll implement this now.

RyBo replied:
Current profit is now calculated. Tested logging in to Miscellania vs. teleporting via Fairy Ring to ensure current amount was being calculated. Also tested changing worker distributions and it seemed to work fine.

trimbe commented on runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java:
```diff
@@ -125,14 +256,24 @@ private void removeKingdomInfobox()
+	private boolean hasCompletedQuests()
 	{
-		return client.getVar(VarPlayer.THRONE_OF_MISCELLANIA) > 0;
+	/*	if (Quest.ROYAL_TROUBLE.getState(client).toString() == "FINISHED")
```
```suggestion
	if (Quest.ROYAL_TROUBLE.getState(client) == QuestState.FINISHED)
```

RyBo commented:
Added tests for calculating the reward values along with the kingdom salaries. Also cleaned up some of the tooltip clutter. More than happy to change anything or add additional tests.

RyBo commented:
I tried to see if there was a varbit associated with how many days since your last rewards collection. However, I don't think this is currently possible so the best you can do is calculate what your current rewards are for the given day.

RyBo commented:
Refactored code to move most of the functionality into the kingdom class. Also changed the tooltip to show all resource yields. It's a bit more cluttered now so changing it to something more minimal might be better.

![kingdomofmiscellania](https://user-images.githubusercontent.com/8285466/53117528-585cd480-3508-11e9-9772-d1e78f75c32f.png)
![kingdomofmiscellaniacalculating](https://user-images.githubusercontent.com/8285466/53117533-5abf2e80-3508-11e9-8b1c-957ac2da300a.png)
